buildtest/menu/show.py

```diff
@@ -1,28 +1,38 @@
 import json
-import sys
+import os
 
-from buildtest.defaults import supported_schemas
 from buildtest.buildsystem.schemas.utils import (
     load_schema,
     get_schemas_available,
     get_schema_fullpath,
+    here,
 )
 
 
 def show_schema_layout(args):
-    """Implements method ``buildtest show schema``"""
-
-    # buildtest show schema --name script
-    if args.name is None:
-        sys.exit("Please provide the name of a schema (e.g., script) to show.")
-
-    # Must be supported
-    if args.name not in supported_schemas:
-        sys.exit(
-            "%s is not a supported schema. Options include %s"
-            % (args.name, "\n".join(supported_schemas))
-        )
-
+    """This method implements command ``buildtest show schema`` which displays json schemas
+       supported by buildtest. The input ``args`` is an instance of argparse class that contains
+       user selection via command line. The method can display the global schema when
+       ``buildtest show schema --global`` is specified or any name schema via
+       ``buildtest show schema --name <NAME>``. The result is an output of the json schema on the console.
+
+       Parameters:
+
+       :param args: instance of argparse class
+       :type args: <class 'argparse.Namespace'>
+       :result: output of json schema on console
+    """
+
+    # implements ``buildtest show schema --global``. Since we can't use args.global we set this to main
+    if args.main:
+        global_schema = os.path.join(here, "global.schema.json")
+        # check if file exists
+        assert global_schema
+        recipe = load_schema(global_schema)
```

Comment:
This should be refactored to get some config schema depending on the args provided, and then one line at the end to print. If you see a pattern that has redundant code and an early return it’s usually an indication that you can implement it more succinctly.

Reply:
I realized that I couldn't use ``get_schemas_available`` https://github.com/shahzebsiddiqui/buildtest-1/blob/schema_updates/buildtest/buildsystem/schemas/utils.py#L104 since it's relied on a specific directory structure like ``script/script-v0.0.1.schema.json``.

My thought was to introduce a separate option ``--global`` instead of using ``--name global``. I could separate out the functionality into separate methods one for name schema (``compiler``, ``script``) and the other for global schema.

Unrelated to this but i see that settings schema is in a different path https://github.com/buildtesters/buildtest/blob/devel/buildtest/settings/settings.schema.json  apart from other schemas and it was intentional.  This means the implementation depends on the directory structure and where all the schemas need to be placed.

I had some other ideas in mind but didn't introduce them yet for example
- Display settings schema https://github.com/buildtesters/buildtest/blob/devel/buildtest/settings/settings.schema.json maybe it could be ``buildtest show schema --settings``. I wasn't sure if it made sense to group this feature in ``buildtest config`` with some option like ``buildtest config schema``
- Add support to fetch development schemas such as https://buildtesters.github.io/schemas/script/script-v0.0.1.schema.json and others.

Reply:
You are misunderstanding. I'm not talking about how it's loaded, but your instantiation of the client here.